src/options.ts

```diff
@@ -0,0 +1,41 @@
+import { SupportOption } from 'prettier';
+
+declare module 'prettier' {
+    interface RequiredOptions extends PluginOptions {}
```

Comment:
That merges prettier's `RequiredOptions` type with whatever the project defines in `PluginOptions`, giving type safe access to the plugin's options without changing any type annotations at specific sites. I think the word required is misleading - AFAICT it means "not partial", since it defines the `Options` as `Partial<RequiredOptions>`.

Here's where `options.sortOrder` gets accessed in [`src/print/index.ts`](https://github.com/UnwrittenFun/prettier-plugin-svelte/pull/22/files#diff-4e442fc9a6c105a9ab3f853a1b9753ceR67):

```typescript
parseSortOrder(options.sortOrder).forEach(p => addParts[p]());
```

In the same file, we didn't need to change the type annotation for prettier's `ParserOptions` type.

```typescript
export function print(path: FastPath, options: ParserOptions, print: PrintFn): Doc {
```

An alternative solution would be to create an `ExtendedParserOptions` type or similar that adds `sortOrder`, and then using that annotation instead of prettier's, but it seemed a lot less clean than just extending prettier's base type in the right spot. This solution would likely require more specialized interfaces in other spots in the future. I haven't used this strategy much but I can only see upsides to it right now.

Reply:
Oh that's pretty cool